### PR TITLE
[mtl] Filament compatibility fixes

### PR DESCRIPTION
**Pull request by kvark**
gfx-filament-pbr-small

There are still occasional asserts on frame acquisition, so leaving #2400 open for now.

PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [ ] tested examples with the following backends:
- [ ] `rustfmt` run on changed code

### CODE REVIEW

**Comment by kvark** [comment]
Problem - RPCS3 regresses with this, it doesn't like that we don't report OUT_OF_DATE on `Oldest` acquiring.

File: src/backend/metal/src/lib.rs [hunk 1]
```diff
@@ -194,18 +193,31 @@ impl Instance {
                 panic!("window does not have a valid contentView");
             }
 
+            let existing: *mut Object = msg_send![view, layer];
             let class = Class::get("CAMetalLayer").unwrap();
```
Comment (grovesNL) [inline src/backend/metal/src/lib.rs]: Unrelated but probably should use [`class!`](https://github.com/SSheldon/rust-objc/blob/master/src/lib.rs#L13) here instead

File: src/backend/metal/src/window.rs [hunk 2]
```diff
@@ -152,16 +164,23 @@ impl Drop for Swapchain {
...
             .inner
             .lock();
-        assert!(!frame.available);
-        frame.available = true;
-        frame.drawable
-            .take()
-            .expect("Drawable has not been acquired!")
+        assert!(!frame.available && frame.linked);
+
+        match frame.drawable.take() {
```
Comment (grovesNL) [inline src/backend/metal/src/window.rs]: Could we use `ok_or` here?

Comment (kvark) [reply src/backend/metal/src/window.rs]: we don't want to set the `available` state unconditionally, so I don't think `ok_or` would help here

File: src/backend/metal/src/lib.rs (hunk 1 above)
Comment (kvark) [reply src/backend/metal/src/lib.rs]: good point! hopefully, this is only a one-time operation

**Comment by kvark** [comment]
thanks for the review!
bors r=grovesNL

**Pull request merged**